### PR TITLE
Begin porting non-dev commands to use AbstractCommand

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
The latest release moved all the dev commands to using `AbstractCommand`. Since we (er, I) haven't seen any reported issues, let's start porting the non-dev commands.

(There's a bonus commit here, where i realized that `require "cli/parser"` should be in `AbstractCommand`, not its subclasses, which no longer reference the parser directly.)

Relates to https://github.com/Homebrew/brew/issues/16814
See also https://github.com/Homebrew/brew/pull/16938 &c.

### CODE REVIEW

File: Library/Homebrew/cmd/help.rb [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+require "abstract_command"
 require "help"
 
 module Homebrew
-  sig { returns(T.noreturn) }
-  def help
-    Help.help
+  module Cmd
+    class HelpCmd < AbstractCommand
+      cmd_args { description "Print help information." }
```
Comment (apainintheneck) [inline Library/Homebrew/cmd/help.rb:10]: Is this really necessary? At a glance it seems like it would never get used.

Comment (dduugg) [reply Library/Homebrew/cmd/help.rb]: Eh, not really. I think I have some follow-up PRs in the queue, for when every command is an `AbstractCommand`, where it simplifies things a bit if every Command also has a `description`. But we can consider this at that time instead.

**Review by apainintheneck — APPROVED** [review]
Great work on porting all of these commands over!

**Review by MikeMcQuaid — APPROVED** [review]
Looks good, thanks again @dduugg!

File: Library/Homebrew/cmd/help.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/help.rb]: (it was necessary, see https://github.com/Homebrew/brew/pull/17038)

**Pull request merged**